src/core/thread/mle_router.cpp

```diff
@@ -2678,7 +2668,22 @@ ThreadError MleRouter::SendChildUpdateRequest(Child *aChild)
     Ip6::Address destination;
     Message *message;
 
+    if ((aChild->GetDeviceMode() & ModeTlv::kModeRxOnWhenIdle) == 0)
```

Comment:
nit: we can use the new `Child` method added by @jwhui here:  `!aChild->IsRxOnWhenIdle()`